In [1]:
def calculate_probability(event_count, total_outcomes):
    return event_count / total_outcomes


# For a coin:
# 1 possible outcome is Heads
# Total possible outcomes = 2
probability_heads = calculate_probability(1, 2)

print("Probability of getting Heads:", probability_heads)

Probability of getting Heads: 0.5


The probability of an event is calculated using:

Probability = Number of favorable outcomes / Total number of outcomes

For a coin toss, there is 1 favorable outcome (Heads) out of 2 possible outcomes.

Probability of Heads = 1 / 2 = 0.5 = 50%

Therefore, the probability of getting Heads is **0.5 or 50%**.

In [3]:
import random

count_6 = 0
total_rolls = 100

for i in range(total_rolls):
    roll = random.randint(1, 6)
    
    if roll == 6:
        count_6 += 1

probability = count_6 / total_rolls

print("Number of times 6 appeared:", count_6)
print("Experimental probability:", probability)

Number of times 6 appeared: 11
Experimental probability: 0.11


I simulated rolling a six-sided die 100 times.

The experimental probability is calculated as:

Experimental Probability = Number of times 6 appeared / Total number of rolls

The theoretical probability of getting a 6 is 1/6 = 0.167.

The experimental probability may be slightly different because the result depends on random rolls.

In [4]:
genres = [
    "Pop", "Rock", "Pop", "Hip-Hop", "Jazz",
    "Pop", "Rock", "Pop", "EDM", "Jazz",
    "Pop", "Rock", "Hip-Hop", "Pop", "Jazz",
    "Rock", "Pop", "EDM", "Pop", "Rock"
]

# Count Pop songs
pop_count = genres.count("Pop")

# Total number of songs
total_songs = len(genres)

# Calculate probability
probability = pop_count / total_songs

print("Number of Pop songs:", pop_count)
print("Total songs:", total_songs)
print("Probability of picking a Pop song:", probability)
print("Probability in percentage:", probability * 100, "%")

Number of Pop songs: 8
Total songs: 20
Probability of picking a Pop song: 0.4
Probability in percentage: 40.0 %


There are 8 Pop songs out of 20 total songs.

Probability = Number of Pop songs / Total number of songs

Probability = 8 / 20 = 0.4

Therefore, the probability of randomly picking a Pop song is **0.4 or 40%**.

In [5]:
def both_order_probability(food_users, dessert_users):
    food_users = set(food_users)
    dessert_users = set(dessert_users)

    # Users who ordered both food and dessert
    both_users = food_users.intersection(dessert_users)

    # Total unique users
    total_users = food_users.union(dessert_users)

    probability = len(both_users) / len(total_users)

    return probability


food_users = ["user1", "user2", "user3", "user4", "user5"]
dessert_users = ["user3", "user4", "user5", "user6", "user7"]

probability = both_order_probability(food_users, dessert_users)

print("Probability of ordering both food and dessert:", probability)
print("Percentage:", probability * 100, "%")


Probability of ordering both food and dessert: 0.42857142857142855
Percentage: 42.857142857142854 %


The users who ordered both food and dessert are found using set intersection.

Food users = 5 users
Dessert users = 5 users
Unique users = 7
Users who ordered both = 3

Probability = 3 / 7 = 0.4286

Therefore, the probability that a randomly selected user ordered both food and dessert is approximately **42.86%**.

In [6]:
# Users who bought a phone
phone_users = 1000

# Users who bought both a phone and headphones
both_users = 300

# Conditional probability
probability = both_users / phone_users

print("Probability of buying headphones given a phone purchase:", probability)
print("Percentage:", probability * 100, "%")

Probability of buying headphones given a phone purchase: 0.3
Percentage: 30.0 %


Conditional probability can help Flipkart recommend products based on a user's previous purchases.

For example, if 1,000 users bought a phone and 300 of those users also bought headphones, then:

P(Headphones | Phone) = 300 / 1000 = 0.30 = 30%

This means there is a 30% probability that a user who bought a phone will also buy headphones. Flipkart could use this information to recommend headphones to users who purchase a phone